impl/src/main/java/io/jsonwebtoken/impl/lang/DefaultCollectionMutator.java

```diff
@@ -46,6 +47,33 @@ private boolean doAdd(E e) {
         return this.collection.add(e);
     }
 
+    public M replace(E e) {
+        if (Objects.isEmpty(e)) {
+            if (e == null)  throw new NullPointerException("Cannot be null.");
+            else            throw new IllegalArgumentException("Cannot be empty.");
+        }
+
+        // Keep a copy of the original, in case remove/add fails, and need to rollback
+        E old = null;
+        for (E element : this.collection) {
+            if (element.equals(e)) {
+                old = element;
+                break;
+            }
+        }
+
+        if (old == null || !this.collection.contains(e)) {
+            String msg = this.getClass() + " does not contain " + e + ".";
+            throw new NoSuchElementException(msg);
+        }
+
+        if (this.collection.remove(e))
+            if (doAdd(e)) changed(); // removed and add successfully, notify changed()
+            else this.collection.add(old); // remove successfully but add failed, add back old (not considered a change)
```

Comment:
This is functionally equivalent for non-ordered collections.
In this example, `e.equals(old) == true`, so it _wouldn't_ need to be replaced.  See above note about changing the method signature.

However, for ordered collections (as is defined in the constructor, the `LinkedHashSet`), this would not be a replace. For example, the collection: `[1,2,3]`
replacing `2` with for `A` _should_ result in `[1,A,3]`, but in the above case would result in `[1,3,A]`

In the implementation above calling `replace(2)` would probably end up with the initial collection of `[1,2,3]` being transformed into `[1,3,2]`